includes/payment-methods/class-klarna-payment-method.php

```diff
@@ -70,6 +70,11 @@ public function get_countries() {
 		if ( in_array( $account_country, $eea_countries, true ) ) {
 			$store_currency = strtoupper( get_woocommerce_currency() );
 
+			// if the store is set to an EU country, but the currency used is not set as a valid EU currency, I guess Klarna shouldn't be eligible.
+			if ( ! isset( $this->limits_per_currency[ $store_currency ] ) ) {
+				return [ 'NONE_SUPPORTED' ];
```

Comment:
I debated for a while what to return here.

If we returned an empty array, it would technically mean that all countries are supported. Not what we want.

So I'm returning an array with one item - a country that doesn't exist.

This kind of solution was initially proposed here (for the currencies, instead of countries): p1742395403731029/1742393258.170269-slack-CU6SYV31A

And I think it fits pretty well.